library/std/Cargo.toml

```diff
@@ -65,6 +65,8 @@ compiler-builtins-mangled-names = ["alloc/compiler-builtins-mangled-names"]
 llvm-libunwind = ["unwind/llvm-libunwind"]
 system-llvm-libunwind = ["unwind/system-llvm-libunwind"]
 
+randomized-layouts = ["alloc/randomized-layouts"]
```

Comment:
Maybe make it a `--cfg` passed using `RUSTFLAGS`? This feature flag is not meant to be set by users and `RUSTFLAGS` applies to all crates without having to add dependent features.

Reply:
That triggers a warning

```
warning: unexpected `cfg` condition name
  --> library/alloc/src/collections/btree/node/tests.rs:97:11
   |
97 | #[cfg(not(randomized_layouts))]
   |           ^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unexpected_cfgs)]` on by default
```

Reply:
you need to add it to the list of `--check-cfg` values in `src/bootstrap`

Reply:
> you need to add it to the list of `--check-cfg` values in `src/bootstrap`

like this: https://github.com/rust-lang/rust/blob/9ba169a73acfa9c9875b76eec09e9a91cc6246df/src/bootstrap/lib.rs#L205

Reply:
thanks

Reply:
Should I make that change to the compiler too or are cargo features ok there?

Reply:
Cargo features under compiler/ seem fine to me, it's not used for -Zbuild-std